<h1 id="heading-1">捕获异常</h1><p>对异常进行处理，称为捕获异常。</p><pre><code class="language-python">x = int(input('输入被除数: '))
y = int(input('输入除数: '))
print(x / y)
</code></pre><p>看起来没问题，如果除数是 0 呢：</p><p>&lt;div class="run"&gt;&lt;/div&gt;</p><pre><code class="language-python">x = 10
y = 0
print(x / y)
</code></pre><p>程序抛出了异常，我们没有捕获它，这个异常的名字叫：</p><pre><code>ZeroDivisionError
</code></pre><p>捕获异常:</p><p>&lt;div class="run"&gt;&lt;/div&gt;</p><pre><code class="language-python">try:
    x = 10
    y = 0
    print(x / y)
except ZeroDivisionError:
    print("除数不能为0!")
</code></pre><h2 id="heading-2">内置的异常类</h2><table class="table bg-cyan-500/30 table-zebra"><tbody><tr class="border-2 border-red-300"><th colspan="1" rowspan="1"><p>类名</p></th><th colspan="1" rowspan="1"><p>描述</p></th></tr><tr class="border-2 border-red-300"><td class="border-2 border-sky-300" colspan="1" rowspan="1"><p>Exception</p></td><td class="border-2 border-sky-300" colspan="1" rowspan="1"><p>几乎所有的异常类都是从它派生而来的</p></td></tr><tr class="border-2 border-red-300"><td class="border-2 border-sky-300" colspan="1" rowspan="1"><p>AttributeError</p></td><td class="border-2 border-sky-300" colspan="1" rowspan="1"><p>引用属性或给它赋值失败时引发</p></td></tr><tr class="border-2 border-red-300"><td class="border-2 border-sky-300" colspan="1" rowspan="1"><p>OSError</p></td><td class="border-2 border-sky-300" colspan="1" rowspan="1"><p>操作系统不能执行指定的任务(如打开文件)时引发，有多个子类</p></td></tr><tr class="border-2 border-red-300"><td class="border-2 border-sky-300" colspan="1" rowspan="1"><p>IndexError</p></td><td class="border-2 border-sky-300" colspan="1" rowspan="1"><p>使用序列中不存在的索引时引发，为 LookupError 的子类</p></td></tr><tr class="border-2 border-red-300"><td class="border-2 border-sky-300" colspan="1" rowspan="1"><p>KeyError</p></td><td class="border-2 border-sky-300" colspan="1" rowspan="1"><p>使用映射中不存在的键时引发，为 LookupError 的子类</p></td></tr><tr class="border-2 border-red-300"><td class="border-2 border-sky-300" colspan="1" rowspan="1"><p>NameError</p></td><td class="border-2 border-sky-300" colspan="1" rowspan="1"><p>找不到名称(变量)时引发</p></td></tr><tr class="border-2 border-red-300"><td class="border-2 border-sky-300" colspan="1" rowspan="1"><p>SyntaxError</p></td><td class="border-2 border-sky-300" colspan="1" rowspan="1"><p>代码不正确时引发</p></td></tr><tr class="border-2 border-red-300"><td class="border-2 border-sky-300" colspan="1" rowspan="1"><p>TypeError</p></td><td class="border-2 border-sky-300" colspan="1" rowspan="1"><p>将内置操作或函数用于类型不正确的对象时引发</p></td></tr><tr class="border-2 border-red-300"><td class="border-2 border-sky-300" colspan="1" rowspan="1"><p>ValueError</p></td><td class="border-2 border-sky-300" colspan="1" rowspan="1"><p>将内置操作或函数用于这样的对象时引发:其类型正确但包含的值不合适</p></td></tr><tr class="border-2 border-red-300"><td class="border-2 border-sky-300" colspan="1" rowspan="1"><p>ZeroDivisionError</p></td><td class="border-2 border-sky-300" colspan="1" rowspan="1"><p>在除法或求模运算的第二个参数为零时引发</p></td></tr></tbody></table><p>无需记忆这些类，遇到时查询文档即可。</p>